enhancements/platform/quota-management/README.md

```diff
@@ -0,0 +1,769 @@
+    - value: "200"
+      selector: {}
+    - value: "20"
+      selector:
+        matchLabels:
+          instanceType: 
+    - value: "5"
+      selector:
+        matchLabels:
+          compute.datumapis.com/location: dfw
+          compute.datumapis.com/instanceType: c3.small
+
+  # 4. Gateway count per project
+  - name: network.datumapis.com/gateways
+    dimensionLabels:
+      - resources.datumapis.com/project
+    buckets:
+    - value: "15"
+      selector: {}
+
+    - value: "3"
+      selector:
+        matchExpressions:
+        - key: compute.datumapis.com/location
+          operator: Exists
+
+```
+
+
+#### `BillingAccount`
+
+```yaml
+
+```
+
+### Quota Operator Controller
+
+A `quota-operator` will be created which implements logic to convert the *intent* of the incoming `ResourceQuotaClaim` object into the *actual allocation* of resources. It maintains accurate usage totals against each `ResourceQuotaGrant` via the `ResourceQuotaGrant.status.usage` field containing the totals of currently allocated resources. The controller is what enforces per-project or per-organization (tenant) resource quota limits that are declared in the `ResourceQuotaGrant` objects.
+
+The reconciliation loop for this controller will contain the following logic:
+
+1. Watches newly generated `ResourceQuotaClaim` objects that were automatically generated by the admission webhook whenever a new resource is created/scaled that will count against the quota.
+2. Validates the `ResourceQuotaClaim` structure and fails early if it is incorrect.
+3. Queries external source (e.g. amberflo) via API to determine the actual usage of the requested resource to accurately inform the decision to be made for the claim.
+4. Makes ultimate decision on whether to `Grant` or `Deny` the claim request based on synced data with the external source.
+5. Reconciles quota thresholds when a `ResourceQuotaGrant` limit changes (e.g. when an administrator raises or lowers quota limits, or the organization undergoes a plan/tier upgrade)
+6. Exposes the resulting status of the request
+
+**Failure Blast Radius**
+
+If the controller is down, new Claims will stall; however workloads are not affected and there is no blocking of writes to the cluster.
+
+### Admission Webhooks
+
+A stateless mutating admission webhook will be created that serves as the initial interaction between the incoming claim request and the `quota-operator` controller. While this is webhook is optional, it provides additional safety of the workflow at claim creation time. 
+
+The admission webhook is responsible for and executes the following steps:
+
+1. Synchronously intercepts incoming requests for resource creation/scaling/deletion that are sent to the `quota-operator` controller.
+2. Automatically creates a `ResourceClaimQuota` object based on the intent of the intercepted request.
+3. Attaches labels and annotations to the object to enable easy claim deletion via a finalizer when workloads are torn down.
```

Comment:
Could you add a sequence diagram showing how this would be used? Since you're using the Instance type as an example above, I think it would be great to understand the flow of instance provisioning and the interaction with the quota system.